flake.nix

```diff
@@ -2,42 +2,61 @@
   description = "A container debugging tool based on FUSE";
 
   inputs = {
-    naersk.url = "github:nmattia/naersk/master";
-    naersk.inputs.nixpkgs.follows = "nixpkgs";
+    fenix.url = "github:nix-community/fenix";
     nixpkgs.url = "github:NixOS/nixpkgs/nixpkgs-unstable";
     utils.url = "github:numtide/flake-utils";
+    flake-compat.url = "github:edolstra/flake-compat";
+    flake-compat.flake = false;
+    nix-filter.url = "github:numtide/nix-filter";
   };
 
-  outputs = { self, nixpkgs, naersk, utils }:
-    (utils.lib.eachDefaultSystem (system: let
-      pkgs = import nixpkgs { inherit system; };
-      naersk-lib = pkgs.callPackage naersk { };
-    in {
-      packages.cntr = pkgs.callPackage ./. {
-        inherit naersk-lib;
-        src = self;
+  outputs =
+    { self
+    , nixpkgs
+    , fenix
+    , utils
+    , flake-compat
+    , nix-filter
+    }@inputs:
+      (
+        utils.lib.eachDefaultSystem (
+          system: let
+            fenix = inputs.fenix.packages.${system};
```

Comment:
GIves us fenix and advantages here over rustc from nixpkgs?

Reply:
If you mean here, there's no specific advantage except that you can use the latest version of rustic. Yeh, I can revert it to use rustc from nixpkgs